packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -124,25 +125,33 @@ function getListeningForDocument(mountAt: any) {
 export function listenTo(
   registrationName: string,
   mountAt: Document | Element,
+  root: Document | Element,
 ) {
-  const isListening = getListeningForDocument(mountAt);
+  const mountAtListeners = getListenerTrackingFor(mountAt);
   const dependencies = registrationNameDependencies[registrationName];
 
   for (let i = 0; i < dependencies.length; i++) {
     const dependency = dependencies[i];
-    if (!(isListening.hasOwnProperty(dependency) && isListening[dependency])) {
+
+    if (!mountAtListeners.hasOwnProperty(dependency)) {
       switch (dependency) {
         case TOP_SCROLL:
-          trapCapturedEvent(TOP_SCROLL, mountAt);
-          break;
+        case TOP_WHEEL:
+          const rootListeners = getListenerTrackingFor(root);
+
+          if (!rootListeners.hasOwnProperty(dependency)) {
+            trapCapturedEvent(dependency, root);
+            rootListeners[dependency] = true;
+          }
+          return;
         case TOP_FOCUS:
         case TOP_BLUR:
```

Comment:
@gaearon you mentioned that touch events were essential (https://github.com/facebook/react/pull/9333#discussion_r212005901), would you like me to add them to the list here?